test/e2e/Tests/Tests/VersioningTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Net;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.Azure.Durable.Tests.DotnetIsolatedE2E;
+
+[Collection(Constants.FunctionAppCollectionName)]
+public class VersioningTests
+{
+    private readonly FunctionAppFixture _fixture;
+    private readonly ITestOutputHelper _output;
+
+    public VersioningTests(FunctionAppFixture fixture, ITestOutputHelper testOutputHelper)
+    {
+        _fixture = fixture;
+        _fixture.TestLogs.UseTestLogger(testOutputHelper);
+        _output = testOutputHelper;
+    }
+
+    [Theory]
+    [InlineData(null)] // Represents a non-versioned case.
+    [InlineData("")] // Non-versioned/empty-versioned case.
+    [InlineData("1.0")]
+    [InlineData("2.0")]
+    public async Task TestVersionedOrchestration_OKWithMatchingVersion(string? version)
+    {
+        string queryString = version == null ? string.Empty : $"?version={version}";
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger("OrchestrationVersion_HttpStart", queryString);
+
+        Assert.Equal(HttpStatusCode.Accepted, response.StatusCode);
+        string statusQueryGetUri = await DurableHelpers.ParseStatusQueryGetUriAsync(response);
+
+        await DurableHelpers.WaitForOrchestrationStateAsync(statusQueryGetUri, "Completed", 30);
+
+        var orchestrationDetails = await DurableHelpers.GetRunningOrchestrationDetailsAsync(statusQueryGetUri);
+        if (version != null)
+        {
+            Assert.Equal($"Version: '{version}'", orchestrationDetails.Output);
+        }
+        else
+        {
+            // The default version (2.0) from the host.json file should've been used here.
+            Assert.Equal("Version: '2.0'", orchestrationDetails.Output);
+        }
+    }
+
+    [Theory]
+    [InlineData(null)] // Represents a non-versioned case.
+    [InlineData("")] // Non-versioned/empty-versioned case.
+    [InlineData("1.0")]
+    [InlineData("2.0")]
+    public async Task TestVersionedSubOrchestration_OKWithMatchingVersion(string? version)
+    {
+        string queryString = version == null ? string.Empty : $"?version={version}";
+        using HttpResponseMessage response = await HttpHelpers.InvokeHttpTrigger("OrchestrationSubVersion_HttpStart", queryString);
+
+        Assert.Equal(HttpStatusCode.Accepted, response.StatusCode);
+        string statusQueryGetUri = await DurableHelpers.ParseStatusQueryGetUriAsync(response);
+
+        await DurableHelpers.WaitForOrchestrationStateAsync(statusQueryGetUri, "Completed", 30);
+
+        var orchestrationDetails = await DurableHelpers.GetRunningOrchestrationDetailsAsync(statusQueryGetUri);
+        if (version != null)
+        {
+            Assert.Equal($"Parent Version: '2.0' | Sub Version: '{version}'", orchestrationDetails.Output);
```

Comment:
I'm confused. Shouldn't both the parent and the child have the same explicitly-specified version?

This is what I was expecting:
```suggestion
            Assert.Equal($"Parent Version: '{version}' | Sub Version: '{version}'", orchestrationDetails.Output);
```

Reply:
Perhaps the test code just needs to be clarified. Instead of labeling the sub-orchestration version parameter as `version` here and in the HTTP trigger function, it should be named `subOrchestrationVersion` or something similar to better clarify how it's intended to be used.

Reply:
Updated it to be `subOrchestrationVersion`. The point of this test is to show we can individually version the sub-orchestration and that it can also take on the default. The parent is only using the default in this case, hence the constant `2.0`.  Hopefully it is clearer now with the new variable :)